server/graphql/v1/TransactionInterface.js

```diff
@@ -236,85 +236,64 @@ const TransactionFields = () => {
-        if (!HostCollectiveId) {
-          const fromCollective = await models.Collective.findByPk(FromCollectiveId);
-          HostCollectiveId = await fromCollective.getHostCollectiveId();
-          // if fromCollective has no host, we try the collective
-          if (!HostCollectiveId) {
-            const collective = await models.Collective.findByPk(CollectiveId);
-            HostCollectiveId = await collective.getHostCollectiveId();
-          }
+
+        const fromCollective = await req.loaders.Collective.byId.load(transaction.FromCollectiveId);
+        if (fromCollective.HostCollectiveId) {
+          return req.loaders.Collective.byId.load(fromCollective.HostCollectiveId);
+        }
+
+        const collective = await req.loaders.Collective.byId.load(transaction.CollectiveId);
+        if (collective.HostCollectiveId) {
+          return req.loaders.Collective.byId.load(fromCollective.HostCollectiveId);
         }
-        return models.Collective.findByPk(HostCollectiveId);
       },
     },
     createdByUser: {
       type: UserType,
       async resolve(transaction, args, req) {
-        // We don't return the user if the transaction has been created by someone who wanted to remain incognito
-        // This is very suboptimal. We should probably record the CreatedByCollectiveId (or better CreatedByProfileId) instead of the User.
-        if (transaction && transaction.getCreatedByUser) {
-          const collective = await transaction.getCollective();
-          const fromCollective = await transaction.getFromCollective();
-          if (fromCollective.isIncognito && !req.remoteUser?.isAdminOfCollectiveOrHost(collective)) {
-            return {};
-          }
-          if (collective.isIncognito && !req.remoteUser?.isAdminOfCollectiveOrHost(fromCollective)) {
-            return {};
-          }
-          return transaction.getCreatedByUser();
+        if (!transaction.CreatedByUserId) {
+          return;
         }
-        return null;
+
+        const [collective, fromCollective] = await Promise.all([
+          req.loaders.Collective.byId.load(transaction.CollectiveId),
+          req.loaders.Collective.byId.load(transaction.FromCollectiveId),
+        ]);
+
+        if (fromCollective.isIncognito && !req.remoteUser?.isAdminOfCollectiveOrHost(collective)) {
+          return {};
+        }
+
+        if (collective.isIncognito && !req.remoteUser?.isAdminOfCollectiveOrHost(fromCollective)) {
+          return {};
+        }
+
+        return req.loaders.User.byId.load(transaction.CreatedByUserId);
       },
     },
     fromCollective: {
       type: CollectiveInterfaceType,
-      resolve(transaction) {
-        // If it's a sequelize model transaction, it means it has the method getFromCollective
-        // otherwise we check whether transaction has 'fromCollective.id', if not we return null
-        if (transaction && transaction.getFromCollective) {
```

Comment:
Now that I see all these *transaction.methodFromTransactionModel*, I wonder if we weren't somehow returning something else (like a raw object) in one of the queries.

Edit: looks like that's the case, as explained here: https://github.com/opencollective/opencollective-api/pull/1578/files#r240981003. But we got rid of this in https://github.com/opencollective/opencollective-api/pull/2384, so these checks are definitely not needed anymore!